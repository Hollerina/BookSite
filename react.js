module.exports = {
  presets: [
    [
      "@babel/preset-react",
      {
        runtime: "automatic",
      },
    ],
  ],
  plugins: [
    ["babel-plugin-transform-minify-gql-template-literals"],
    [
      "@emotion",
      {
        sourceMap: false,
        autoLabel: "always",
        labelFormat: "[local]--[filename]",
      },
    ],
  ],
};
