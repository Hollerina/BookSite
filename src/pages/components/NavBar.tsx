import { SignedIn, UserButton } from "@clerk/clerk-react"
import { useLocation } from "react-router-dom"
import styled from "@emotion/styled"
import { theme } from "../../styling/Themeing"

const NavBarContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  background-color: ${theme.colours.accentColours.Glaucous};
  align-items: center;
  padding-top: 8px;
  border-bottom: ${theme.colours.blacks.smokeyBlack} 1px solid;
`

const AnchorContainer = styled.div`
  display: flex;
`

const ActiveAnchor = styled.a`
  background-color: ${theme.colours.backgroundColours.babyPowder};
  color: black;
  padding: 11px 12px;
  margin-left: 5px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border-left: ${theme.colours.blacks.smokeyBlack} solid 1px;
  border-right: ${theme.colours.blacks.smokeyBlack} solid 1px;
  border-top: ${theme.colours.blacks.smokeyBlack} solid 1px;
  border-bottom-color: transparent;
  margin-bottom: -1px;
  text-decoration: none;
  width: 157px;
  text-align: center;
`

const InactiveAnchor = styled.a`
  background-color: ${theme.colours.accentColours.Glaucous};
  color: ${theme.colours.backgroundColours.babyPowder};
  padding: 11px 12px;
  margin-left: 5px;
  text-decoration: none;
  width: 157px;
  text-align: center;

  &:hover {
    text-decoration: underline;
    text-underline-offset: 5px;
    text-decoration-thickness: 2px;
    text-decoration-color: ${theme.colours.backgroundColours.isabelline};
  }
`

const ProfileContainer = styled.div``

const Navbar = () => {
  const location = useLocation()

  return (
    <NavBarContainer>
      <AnchorContainer>
        {location.pathname === "/profile" ? (
          <ActiveAnchor href="/profile" target="_self" className="active">
            Profile
          </ActiveAnchor>
        ) : (
          <InactiveAnchor href="/profile" target="_self" className="inactive">
            Profile
          </InactiveAnchor>
        )}
        {location.pathname === "/profile/currently-reading" ? (
          <ActiveAnchor
            href="/profile/currently-reading"
            target="_self"
            className="active"
          >
            Currently Reading
          </ActiveAnchor>
        ) : (
          <InactiveAnchor
            href="/profile/currently-reading"
            target="_self"
            className="inactive"
          >
            Currently Reading
          </InactiveAnchor>
        )}
        {location.pathname === "/profile/read-books" ? (
          <ActiveAnchor
            href="/profile/read-books"
            target="_self"
            className="active"
          >
            Read Books
          </ActiveAnchor>
        ) : (
          <InactiveAnchor
            href="/profile/read-books"
            target="_self"
            className="inactive"
          >
            Read Books
          </InactiveAnchor>
        )}
      </AnchorContainer>
      <ProfileContainer>
        <SignedIn>
          <UserButton />
        </SignedIn>
      </ProfileContainer>
    </NavBarContainer>
  )
}

export default Navbar
