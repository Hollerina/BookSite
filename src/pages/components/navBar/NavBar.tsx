import { SignedIn, UserButton } from "@clerk/clerk-react"
import styled from "@emotion/styled"
import { theme } from "../../../styling/Themeing"
import { maxLg } from "../../../styling/Responsive"

import MobileNavLinks from "./components/MobileNavLinks"
import DesktopNavLinks from "./components/DesktopNavLinks"

const HeaderTitle = styled.h1`
  color: ${theme.colours.backgroundColours.babyPowder};
  margin-bottom: -1px;
  margin-top: 0;
  padding: 0 20px;
`

const DesktopContainer = styled.div`
  display: flex;
`

const NavBarContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  background-color: ${theme.colours.accentColours.Glaucous};
  align-items: center;
  padding-top: 8px;
  border-bottom: ${theme.colours.blacks.smokeyBlack} 1px solid;

  ${maxLg("padding-top: 0px")}
`

const ProfileContainer = styled.div`
  padding: 10px;
  display: flex;

  justify-content: center;
  align-items: center;
`

const NavBar = () => {
  return (
    <NavBarContainer>
      <MobileNavLinks />
      <DesktopContainer>
        <HeaderTitle>BookSite</HeaderTitle>
        <DesktopNavLinks />
      </DesktopContainer>
      <ProfileContainer>
        <SignedIn>
          <UserButton />
        </SignedIn>
      </ProfileContainer>
    </NavBarContainer>
  )
}

export default NavBar
