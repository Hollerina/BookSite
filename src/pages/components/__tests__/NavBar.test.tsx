import { fireEvent, getByText, render } from "@testing-library/react";
import NavBar from "../NavBar";
import "@testing-library/jest-dom";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { ClerkProvider } from "@clerk/clerk-react";
import { screen } from "@testing-library/react";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <MemoryRouter>
      <ClerkProvider
        routerPush={(to) => {}}
        routerReplace={(to) => {}}
        publishableKey={
          "pk_test_YmVjb21pbmctYmx1ZWdpbGwtNjEuY2xlcmsuYWNjb3VudHMuZGV2JA"
        }
      >
        {children}
      </ClerkProvider>
    </MemoryRouter>
  );
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/profile",
  }),
}));

describe("Navbar", () => {
  //Snapshot doesn't render with CSS
  it("should match the snapshot", () => {
    const { container } = render(<NavBar />, { wrapper: AllTheProviders });

    expect(container).toMatchSnapshot();
  });
  it.each([
    ["/profile", "Profile"],
    ["/profile/currently-reading", "Currently Reading"],
    ["/profile/read-books", "Read Books"],
  ])("should have a href %s when %s is clicked", (hrefLink, navbarTitle) => {
    render(<NavBar />, { wrapper: AllTheProviders });
    const link = screen.getByText(navbarTitle);
    fireEvent.click(link);

    expect(screen.getByRole("link", { name: navbarTitle })).toHaveAttribute(
      "href",
      hrefLink
    );
  });
});
