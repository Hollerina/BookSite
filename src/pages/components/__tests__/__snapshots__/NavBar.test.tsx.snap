// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
<div>
  <div
    class="sc-braxZu bzdGCH"
  >
    <div
      class="sc-gJhJTp eUaePN"
    >
      <a
        class="sc-fQpRED eiKtPW"
        href="/profile"
        target="_self"
      >
        Profile
      </a>
      <a
        class="sc-fQpRED eiKtPW"
        href="/profile/currently-reading"
        target="_self"
      >
        Currently Reading
      </a>
      <a
        class="sc-fQpRED eiKtPW"
        href="/profile/read-books"
        target="_self"
      >
        Read Books
      </a>
    </div>
    <div
      class="sc-dsLQwm"
    />
  </div>
</div>
`;
